mssql/testcontainers/mssql/__init__.py

```diff
@@ -39,7 +54,21 @@ def _configure(self) -> None:
         self.with_env("ACCEPT_EULA", 'Y')
 
     def get_connection_url(self) -> str:
-        return super()._create_connection_url(
+        url = super()._create_connection_url(
             dialect=self.dialect, username=self.SQLSERVER_USER, password=self.SQLSERVER_PASSWORD,
             db_name=self.SQLSERVER_DBNAME, port=self.port_to_expose
         )
+        if self.dialect == "mssql+pyodbc":
+            import pyodbc
+            import re
+            r = re.compile('ODBC Driver \d{1,2} for SQL Server')
+            # We sort drivers in reversed order to get the latest
+            drivers = sorted(list(filter(r.match, pyodbc.drivers())), reverse=True)
+
+            if len(drivers) > 0:
+                driver_str = drivers[0].replace(" ", "+")
```

Comment:
My hunch is that this won't work because `sorted` is an iterable that can't be indexed. We likely also need to parse the version as an integer because, using string comparison, version 8 is greater than version 10. E.g.,

```python
>>> max(['ODBC Driver 10 for SQL Server', 'ODBC Driver 8 for SQL Server'])
'ODBC Driver 8 for SQL Server'
```